### ISSUE TITLE
[CascadingCallWrapping] False negative

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#5042]
## Expected Behavior

```kotlin
foo.bar(
  ""
).bar2()
  .bar3()
```

Should raise an exception because `.bar2()` doesn't follow the chain correctly. Or this should be handled by a different rule? I'm not 100% sure about this.

## Observed Behavior
The rule doesn't complain about that code

## Context
I had some similar code some of them was raising this issue and some of them not. The reason was that this code raise an issue:

```kotlin
foo
  .bar(
    ""
  ).bar2()
```

And to me it was a bit confusing why one raised the issue and the other doesn't.

## Your Environment
* Version of detekt used: 1.21.0-RC2

**Comment by gouri-panda** [comment]
@BraisGabin I can investigate this issue 👍

**Comment by gouri-panda** [comment]
``` kt
foo.bar(
  ""
).bar2()
  .bar3()
```
@BraisGabin I guess  it doesn't raise an exception when the first operator in this  case `bar()` is not inlined a new line that's why it didn't raise an exception.
```kt
foo
  .bar(
    ""
  ).bar2()
```
In the second case `bar()` aligned with a new line that's why it raises an exception in `bar2()`. I guess it's a bug. Should we raise an exception in the first case that `bar()` and `bar2()` should wrap to new lines?

**Comment by BraisGabin** [comment]
I don't fully agree with that change in the rule but seems I'm the only one 😅. So yes. This issue is a "works as expected".
